src/Data/Conduit/Tar.hs

```diff
@@ -789,6 +796,24 @@ tarEntries = do
     yield terminatorBlock
     return $ offset + fromIntegral (S.length terminatorBlock)
 
+-- | Construct a directory with modification time set to current time and the owner set to @root@.
+--
+-- @since 0.3.3
+makeDirectory :: MonadIO m => FilePath -> m FileInfo
```

Comment:
I think a different name would be better. Currently, it sounds almost identical to `createDirectory`.